pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -391,6 +398,15 @@ impl<T: Config> Pallet<T> {
 		let mut state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
 		let mut updated_scheduled_requests = vec![];
 
+		// backwards compatible handling for DelegatorStatus::Leaving
+		#[allow(deprecated)]
+		if matches!(state.status, DelegatorStatus::Leaving(_)) {
+			state.status = DelegatorStatus::Active;
+			<DelegatorState<T>>::insert(delegator.clone(), state.clone());
+			Self::deposit_event(Event::DelegatorExitCancelled { delegator });
+			return Ok(().into());
```

Comment:
```suggestion
```

why emit this event and return success early here without cancelling all requests further below?

Reply:
The reason is that these delegators have no scheduled revoke requests, since this is from previous behavior where `schedule_leave_delegators` only changed the `status` without inserting any requests.

Reply:
These delegators can schedule revoke requests before they call this?

Reply:
I mean I am a delegator, my state is `Leaving`, can I schedule revoke requests?

Reply:
Even if you can, I think because you have the status "leaving" it means we should treat it the same as it was before, by 
changing your status only, and not touching your revokes.

post-upgrade schedule_leave should not change the status to "Leaving" anymore so those won't have trouble with this part of the code

Reply:
Yes, one can always manually schedule revokes but as per RT1600 we translated `Leaving` to `all delegations have a scheduled revoke`. Thus cancel would only return `Ok` iff all delegations have revoke, otherwise `DelegatorNotLeaving`.

For delegators before RT1600, 
  if having `0` revokes (most likely)  -> simply change the status to `Active`
  if having `some` revokes  -> old behavior simply changes the status to `Active` without touching scheduled requests, new behavior will preserve that since they can manually cancel them.
  if having `all` revokes -> someone went overboard and created revokes en masse, old behavior would leave them untouched, new behavior preserves that